pkg/nuctl/command/export.go

```diff
@@ -0,0 +1,306 @@
+	cmd.PersistentFlags().BoolVar(&commandeer.noScrub, "no-scrub", false, "Don't scrub function data")
+
+	commandeer.cmd = cmd
+
+	return commandeer
+}
+
+func (e *exportFunctionCommandeer) renderFunctions(functions []platform.Function, format string, writer io.Writer) error {
+
+	// iterate over each function and make sure it's initialized
+	// TODO: parallelize
+	for _, function := range functions {
+		if err := function.Initialize(nil); err != nil {
+			return err
+		}
+	}
+
+	rendererInstance := renderer.NewRenderer(writer)
+
+	switch format {
+	case OutputFormatYAML:
+		return e.renderFunctionConfig(functions, rendererInstance.RenderYAML)
+	case OutputFormatJSON:
+		return e.renderFunctionConfig(functions, rendererInstance.RenderJSON)
+	}
+
+	return nil
+}
+
+func (e *exportFunctionCommandeer) renderFunctionConfig(functions []platform.Function, renderer func(interface{}) error) error {
+	if len(functions) == 1 {
+		functionConfig := functions[0].GetConfig()
+		functionconfig.PrepareFunctionForExport(functionConfig, e.noScrub)
+		if err := renderer(functionConfig); err != nil {
+			return errors.Wrap(err, "Failed to render function config")
+		}
+		return nil
+	}
+
+	functionConfigs := map[string]*functionconfig.Config{}
+	for _, function := range functions {
+		functionConfig := function.GetConfig()
+		functionconfig.PrepareFunctionForExport(functionConfig, e.noScrub)
+		functionConfigs[functionConfig.Meta.Name] = functionConfig
+	}
+	if err := renderer(functionConfigs); err != nil {
+		return errors.Wrap(err, "Failed to render function config")
+	}
+
+	return nil
+}
+
+type exportProjectCommandeer struct {
+	*exportCommandeer
+	getProjectsOptions platform.GetProjectsOptions
+	output             string
+}
+
+func newExportProjectCommandeer(exportCommandeer *exportCommandeer) *exportProjectCommandeer {
+	commandeer := &exportProjectCommandeer{
+		exportCommandeer: exportCommandeer,
+	}
+
+	cmd := &cobra.Command{
+		Use:     "project name",
+		Aliases: []string{"proj"},
+		Short:   "Export project with all it's functions and function events",
+		RunE: func(cmd *cobra.Command, args []string) error {
+
+			// if we got positional arguments
+			if len(args) != 0 {
+
+				// second argument is a resource name
+				commandeer.getProjectsOptions.Meta.Name = args[0]
+			}
+
+			// initialize root
+			if err := exportCommandeer.rootCommandeer.initialize(); err != nil {
+				return errors.Wrap(err, "Failed to initialize root")
+			}
+
+			// get namespace
+			commandeer.getProjectsOptions.Meta.Namespace = exportCommandeer.rootCommandeer.namespace
+
+			projects, err := exportCommandeer.rootCommandeer.platform.GetProjects(&commandeer.getProjectsOptions)
+			if err != nil {
+				return errors.Wrap(err, "Failed to get projects")
+			}
+
+			if len(projects) == 0 {
+				cmd.OutOrStdout().Write([]byte("No projects found")) // nolint: errcheck
```

Comment:
also here, differentiate between get detail and get list, on the former return with a not found error